test/sharness/lib/test-lib.sh

```diff
@@ -353,3 +355,38 @@ test_check_peerid() {
 	}
 }
 
+# A workaround for https://github.com/ipfs/go-ipfs/issues/1941 ‘sharness suite:
+# ipfs name publish: “Error: failed to find any peer in table”’.  Certain
+# commands fail whenever the daemon is running with no peers.
+# test_expect_success_1941 runs test_expect_failure if that is the case and
+# test_expect_success otherwise.  As soon as #1941 is fixed, all invocations of
+# test_expect_success_1941 can be replaced with test_expect_success and this
+# function can be removed.
```

Comment:
> Previously there were duplicated blocks of tests where the succeeding (offline) tests used test_expect_success and the failing (online) tests used test_expect_failure but that was not acceptable. 

agreed.

> Now both use the same test_expect_success_1941 which is temporary and will be globally replaced with test_expect_success as soon as #1941 is fixed.

maybe #1941 should be fixed beforehand then. what will it take? cc @whyrusleeping 

@ion1 my reluctance is that once something is merged in, it often takes a very long time to remove it. For example, the writable gateway was working a long time ago. it was briefly temporarily removed, but that "temporary" lasted many months.

in this case, it may be ok as the code should do the right thing.

in any case, can this even merge on master? @whyrusleeping has asked for a freeze and for all new code to be added to dev0.4.0. If @whyrusleeping wants it to be on 0.4.0, then should

- [ ] rebase this on dev0.4.0
- [ ] see what it will take to fix #1941